redfish/provider/resource_redfish_virtual_media.go

```diff
@@ -1,26 +1,456 @@
+					"NFS",
+					"SCP",
+					"TFTP",
+					"OEM",
+				}...),
+			},
+		},
+		"write_protected": schema.BoolAttribute{
+			Optional:            true,
+			Computed:            true,
+			Description:         "Indicates whether the remote device media prevents writing to that media.",
+			MarkdownDescription: "Indicates whether the remote device media prevents writing to that media.",
+			Default:             booldefault.StaticBool(true),
+		},
+	}
+}
+
+// Schema defines the schema for the resource.
+func (*virtualMediaResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Resource for managing virtual media.",
+		Version:             1,
+		Attributes:          VirtualMediaSchema(),
+	}
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *virtualMediaResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	tflog.Trace(ctx, "resource_virtual_media create : Started")
+	// Get Plan Data
+	var plan models.VirtualMedia
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	// Validate image extension
+	image := plan.Image.ValueString()
+	if !strings.HasSuffix(image, ".iso") && !strings.HasSuffix(image, ".img") {
+		resp.Diagnostics.AddError(mountError, "Unable to Process the request. Image extension should be .iso or .img")
+		return
+	}
+	// Validate transfer method
+	if plan.TransferMethod.ValueString() == "Upload" {
+		resp.Diagnostics.AddError(mountError, "Unable to Process the request. TransferMethod upload is not supported.")
+		return
+	}
+	virtualMediaConfig := redfish.VirtualMediaConfig{
+		Image:                image,
+		Inserted:             plan.Inserted.ValueBool(),
+		TransferMethod:       plan.TransferMethod.ValueString(),
+		TransferProtocolType: plan.TransferProtocolType.ValueString(),
+		WriteProtected:       plan.WriteProtected.ValueBool(),
+	}
+	// Get service
+	service, err := NewConfig(r.p, &plan.RedfishServer)
+	if err != nil {
+		resp.Diagnostics.AddError(serviceError, err.Error())
+		return
+	}
+	// Get Systems details
+	systems, err := service.Systems()
+	if err != nil {
+		resp.Diagnostics.AddError("Error when retrieving systems", err.Error())
+		return
+	}
+	if len(systems) == 0 {
+		resp.Diagnostics.AddError("There is no system available", err.Error())
+		return
+	}
+	// Get virtual media collection
+	virtualMediaCollection, err := systems[0].VirtualMedia()
+	if err != nil {
+		resp.Diagnostics.AddError("Couldn't retrieve virtual media collection from redfish API", err.Error())
+		return
+	}
+	if len(virtualMediaCollection) != 0 {
+		for index := range virtualMediaCollection {
+			virtualMedia, err := insertMedia(virtualMediaCollection[index].ID, virtualMediaCollection, virtualMediaConfig, service)
+			if err != nil {
+				resp.Diagnostics.AddError("Error: %s", err.Error())
```

Comment:
String formatting won't work here. Suggest something like
```suggestion
				resp.Diagnostics.AddError("Error while inserting virtual media", err.Error())
```

Reply:
done